### PR TITLE
box: export more symbols

### LINKED ISSUE #10378: Export several symbols
These symbols are available in lua API, but not in C API.
There's a demand for having them in C API too.
```
box_info_lsn
box_is_ro
box_ro_reason
box_wait_ro
```

### PR DESCRIPTION
**Pull request by alyapunov**
box_info_lsn
box_is_ro
box_ro_reason
box_wait_ro

Closes #10378

NO_CHANGELOG=minor change

@TarantoolBot document
Title: document 4 more symbols in public C API

box_info_lsn
box_is_ro
box_ro_reason
box_wait_ro

Their meaning is identical to lua methods:

box.info.lsn
box.info.ro
box.info.ro_reason
box.wait_ro

### CODE REVIEW

File: src/box/box.h [hunk 1]
```diff
@@ -188,11 +188,10 @@ box_check_writable(void);
 void
 box_set_ro(void);
 
-bool
-box_is_ro(void);
+/** \cond public */
```
[the remark this answers is not in the document]
Comment (alyapunov) [reply src/box/box.h:191]: Thaks, it was ugly cherry-pick from 2.11.3 )

**Review by Totktonada — APPROVED** [review]
I've no objections.

Remarks:

* We can add the `API_EXPORT` macro to the prototypes for unification (see 8e9558475082bdf8383f239f19bfb9dcc4518510).
* `box_wait_ro` name may be confusing (it allows to wait for the RW mode), but it is clear after reading the documentation comment. So, just noted.

File: src/box/box.cc [hunk 2]
```diff
@@ -4212,6 +4212,24 @@ box_iproto_override(uint32_t req_type, iproto_handler_t handler,
 	return iproto_override(req_type, handler, destroy, ctx);
 }
 
+int64_t
```
[the remark this answers is not in the document]
Comment (alyapunov) [reply src/box/box.cc:4215]: done

[the remark this answers is not in the document]
Comment (sergepetrenko) [reply src/box/box.cc:4215]: Thanks!

**Review by sergepetrenko — APPROVED** [review]
Thanks for the patch!

**Comment by sergepetrenko** [comment]
Backports:
3.1 https://github.com/tarantool/tarantool/pull/10428
2.11 https://github.com/tarantool/tarantool/pull/10427

**Pull request merged**